### ISSUE TITLE
Fix exit code when nothing is passed to stylelint command

### ISSUE THREAD

**Issue by jlurena** [stylelint/stylelint#5150]
> What is the problem you're trying to solve?

I am running stylelint on Jenkins. I pull the diffs of Base PR and Current PR and then pass the filenames to stylelint. 

Sometimes the diff will be nil or an empty string.

`git diff origin/${CHANGE_TARGET}... --name-only | egrep '\.(scss|css)$' | xargs`

I pass the above to stylelint like so

```
stylelint --aei $(git diff origin/${CHANGE_TARGET} --name-only | egrep '\.(scss|css)$' | xargs)
```

It fails...

> What solution would you like to see?

If no glob/filename is passed, lint check fails. 

**It should not fail**. (ie eslint returns with an exit code of 0 if nothing is passed)

stylelint

```
$ stylelint
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

eslint

```
$ eslint
$ eslint
✨  Done in 0.50s.
```

**Comment by jeddy3** [comment]
@jlurena Thanks for the request, for using the template and clearly explaining your use case.

It would really help us if you dug out historical issues related to exit codes. I feel like we've gone back and forth on this over the years, and I'd like to make sure we don't inadvertently break someone else's use case when addressing yours.

**Comment by jlurena** [comment]
> @jlurena Thanks for the request, for using the template and clearly explaining your use case.
> 
> It would really help us if you dug out historical issues related to exit codes. I feel like we've gone back and forth on this over the years, and I'd like to make sure we don't inadvertently break someone else's use case when addressing yours.

Its not so much about exit codes, but more about not failing if no param is passed. Maybe allow a flag to be passed that won't throw an error if an empty no param s passed?

```
stylelint --allow-no-params
```

Or something along those lines.

**Comment by jlurena** [comment]
Or better yet, just print out the usage information instead of exiting with an error code.

```
stylelint 
# Prints out usage info found in `stylelint -h`
```

**Comment by jeddy3** [comment]
> just print out the usage information instead of exiting with an error code.

It should do that already.

```
jeddy3@mac stylelint.io % npx stylelint              

  A mighty, modern CSS linter.

  Usage: stylelint [input] [options]

  Input: Files(s), glob(s), or nothing to use stdin.

    If an input argument is wrapped in quotation marks, it will be passed to
    globby for cross-platform glob support. node_modules are always ignored.
    You can also pass no input and use stdin, instead.

  ..
```

Are you not seeing this behaviour?

It also sets the exit code to `2`, though. Whereas `npx stylelint --help` does not.

We should probably make these consistent and exit with `0`. Would that work for your use case?

**Comment by jlurena** [comment]
> > just print out the usage information instead of exiting with an error code.
> 
> It should do that already.
> 
> ```
> jeddy3@mac stylelint.io % npx stylelint              
> 
>   A mighty, modern CSS linter.
> 
>   Usage: stylelint [input] [options]
> 
>   Input: Files(s), glob(s), or nothing to use stdin.
> 
>     If an input argument is wrapped in quotation marks, it will be passed to
>     globby for cross-platform glob support. node_modules are always ignored.
>     You can also pass no input and use stdin, instead.
> 
>   ..
> ```
> 
> Are you not seeing this behaviour?
> 
> It also sets the exit code to `2`, though. Whereas `npx stylelint --help` does not.
> 
> We should probably make these consistent and exit with `0`. Would that work for your use case?

Hey sorry for the late reply, but yes, that is what I am suggesting.

`npx stylelint` -> should not throw an error

**Comment by jeddy3** [comment]
I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

**Comment by jlurena** [comment]
> I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

Should it display the usage or just not do anything?

**Comment by jeddy3** [comment]
Display the usage as it currently does.

**Comment by Ninglo** [comment]
I found that executing npx stylelint under 15.10.2 would hang the process, which would cause my CI process to never end, has something changed here directly?
```bash
➜  testProject git:(style/new-lint) ✗ npx stylelint --version
15.10.2
➜  testProject git:(style/new-lint) ✗ npx stylelint


```

**Comment by ybiquitous** [comment]
@Ninglo Sorry for the inconvenience. We've recently fixed the regression with PR #7131. It will be released with the next major version 16.0.0.
